payment/payment_test.go

```diff
@@ -29,3 +29,12 @@ func TestSegwitAddress(t *testing.T) {
 		t.Errorf("TestSegwitAddress: error when encoding segwit")
 	}
 }
+
+func TestScriptHash(t *testing.T) {
+	_, publicKey := btcec.PrivKeyFromBytes(btcec.S256(), privateKeyBytes)
+
+	pay := payment.FromPublicKey(publicKey, &network.Regtest)
```

Comment:
This is not semantically correct. The Hash, in this case, should not represent the hash of a supplied public key, but a redeem script. https://en.bitcoin.it/wiki/BIP_0016

The correct way to do this would be to implement first a FromRedeem factory function, where you could pass then a PubKeyHash or WitnessPubKeyHash

Reply:
Something like 

```go
pkh := payment.FromPublicKey(publicKey,Regtest)
psh := payment.FromRedeem(pkh,Regtest)
address := psh.ScriptHash()
```

Reply:
ScriptHash should use the `Hash` field, that should be valorized by the FromReedem function using the `Script`

Reply:
This task #11, is dependent on task #9 cause ScriptHash is invalid without the creation of Payment type using Factory func FromPayment(previously named FromRedeem).
Should I add changes that were meant to be in task #9 in this PR?
@tiero please advise

Reply:
@sekulicd  Yes it's correct. Better way is to close this PR, open a new one for task #9 and only then reopen a new one for task #11. Thanks

@altafan For the future: this task should have been created and/or assigned only AFTER the task #9 got merged in master, in order to not have dependency with other tasks.